components/indexPage_test.tsx

```diff
@@ -0,0 +1,204 @@
+                    /
+                  </a>
+                </span>
+                <span class="tw-1xinsej">
+                  <p>an example module</p>
+                </span>
+              </label>
+              <table class="m-4">
+                <tbody>
+                  <tr>
+                    <td colSpan={2} class="py-2">
+                      <a
+                        href="/https://deno.land/x/tst//mod.ts"
+                        class="pr-4 text-blue-800 dark:text-blue-300 hover:underline"
+                      >
+                        /mod.ts
+                      </a>
+                      <span class="tw-1xinsej">
+                        <p>an example module</p>
+                      </span>
+                    </td>
+                  </tr>
+                </tbody>
+              </table>
+            </div>
+            <div class="tw-n0dh8j" id="group_lib">
+              <input
+                type="checkbox"
+                id="id_lib"
+                checked
+                class="hidden"
+                aria-expanded="false"
+                aria-controls="group_lib"
+              />
+              <label
+                for="id_lib"
+                class="block p-2 border-b border-gray-400 dark:border-gray-600 cursor-pointer"
+              >
+                <img
+                  class="inline rotate-90 dark:filter dark:invert mr-2"
+                  height="24"
+                  width="24"
+                  src="/static/arrow_right.svg"
+                />
+                <span class="mr-4">
+                  <a
+                    href="/https://deno.land/x/tst//lib"
+                    class="text-blue-800 dark:text-blue-300 hover:underline"
+                  >
+                    /lib
+                  </a>
+                </span>
+              </label>
+              <table class="m-4">
+                <tbody>
+                  <tr>
+                    <td colSpan={2} class="py-2">
+                      <a
+                        href="/https://deno.land/x/tst//lib/mod.ts"
+                        class="pr-4 text-blue-800 dark:text-blue-300 hover:underline"
+                      >
+                        /lib/mod.ts
+                      </a>
+                    </td>
+                  </tr>
+                </tbody>
+              </table>
+            </div>
+          </div>
+        </article>
+      </div>
+    );
+    sheet.reset();
+    const requestUrl = new URL(
+      "https://doc.deno.land/https://deno.land/x/tst/",
+    );
+    const actual = renderSSR(
+      <IndexPage
+        base="/https://deno.land/x/tst/"
+        description="test description"
+        path=""
+        requestUrl={requestUrl}
+        url="https://deno.land/x/tst/"
+      >
+        {indexStructure}
+      </IndexPage>,
+    )
+      .replaceAll("\n", "");
+    const expected = renderSSR(<Expected />).replaceAll("\n", "");
+    assertEquals(actual, expected);
```

Comment:
FYI we have new [`assertSnapshot`](https://deno.land/manual@main/testing/snapshot_testing) API available which could potentially make these kind of tests nicer